### DISCUSSION TITLE
question about the field  default legend

### DISCUSSION THREAD

**Discussion by liuchyi** [grafana/grafana-json-datasource#57] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/6964441/106991045-6e750580-67b0-11eb-834a-e2a35f85db47.png)

hello, how to rename the field  default legend in the pic above
i want change 0, 1, 2 legend by myself

**Comment by marcusolsson** [comment, marked as answer]
There are two ways:

- Use the Organize fields transform in the Transform tab
- Use a field override to override the Display name

Let me know if that doesn't work!

**Comment by liuchyi** [comment]
thanks, awesome，the first one work perfect!
and another question, how to ignore or replace NaN value  that unable to graph data
![image](https://user-images.githubusercontent.com/6964441/107004945-fd424c00-67c9-11eb-8e7f-8c52fad10183.png)

![image](https://user-images.githubusercontent.com/6964441/107004965-0501f080-67ca-11eb-8fab-166fa6e7c874.png)

**Comment by marcusolsson** [comment]
It looks like the fields are detected as strings. I think it might work if you change the Type from 'Auto' to 'Number', to force the type.

**Reply by liuchyi** [reply]
good answer!
